src/mmap.rs

```diff
@@ -728,6 +728,57 @@ mod tests {
     type GuestRegionMmap = super::GuestRegionMmap<()>;
     type MmapRegion = super::MmapRegion<()>;
 
+    #[cfg(not(all(feature = "xen", unix)))]
+    mod defs {
+        use super::*;
+
+        pub fn as_range(
+            addr: GuestAddress,
+            size: usize,
+            file: Option<FileOffset>,
+        ) -> (GuestAddress, usize, Option<FileOffset>) {
```

Comment:
We should just have a specialised return type for `as_range` instead. This way we also don't need to duplicate these defines. The type for xen will have the 5 fields, while the other one will have 3. Then the `as_range` function will have the same definition in both cases.